cli-output/src/cli_output.rs

```diff
@@ -610,25 +610,63 @@ fn show_votes_and_credits(
         return Ok(());
     }
 
-    writeln!(f, "Recent Votes:")?;
-    for vote in votes {
-        writeln!(f, "- slot: {}", vote.slot)?;
-        writeln!(f, "  confirmation count: {}", vote.confirmation_count)?;
-    }
-    writeln!(f, "Epoch Voting History:")?;
+    // Existence of this should guarantee the occurrence of vote truncation
+    let newest_history_entry = epoch_voting_history.iter().rev().next();
+
     writeln!(
         f,
-        "* missed credits include slots unavailable to vote on due to delinquent leaders",
+        "{} Votes (using {}/{} entries):",
+        (if newest_history_entry.is_none() {
+            "All"
+        } else {
+            "Recent"
+        }),
+        votes.len(),
+        MAX_LOCKOUT_HISTORY
     )?;
-    for entry in epoch_voting_history {
+
+    for vote in votes.iter().rev() {
+        writeln!(
+            f,
+            "- slot: {} (confirmation count: {})",
+            vote.slot, vote.confirmation_count
+        )?;
+    }
+    if let Some(newest) = newest_history_entry {
+        writeln!(
+            f,
+            "- ... (truncated older {} votes, which was rooted and became credits)",
+            newest.credits
+        )?;
+    }
+
+    if !epoch_voting_history.is_empty() {
+        writeln!(
+            f,
+            "{} Epoch Voting History (using {}/{} entries):",
+            (if epoch_voting_history.len() < MAX_EPOCH_CREDITS_HISTORY {
+                "All"
+            } else {
+                "Recent"
+            }),
+            epoch_voting_history.len(),
+            MAX_EPOCH_CREDITS_HISTORY
+        )?;
+        writeln!(
+            f,
+            "* missed credits include slots unavailable to vote on due to delinquent leaders",
+        )?;
+    }
+
+    for entry in epoch_voting_history.iter().rev() {
```

Comment:
NIT: Maybe give this block the single-line treatment as well?

Reply:
well, I rather keep this as-is. Maybe, the single-line style condenses too much information in a single line.